src/mbgl/style/paint_property.hpp

```diff
@@ -39,8 +39,12 @@ class PaintProperty {
         return *this;
     }
 
+    bool isUndefined() const {
+        return values.find(ClassID::Default) == values.end();
+    }
+
     const PropertyValue<T>& get() const {
-        return values.at(ClassID::Default);
+        return isUndefined() ? values.at(ClassID::Fallback) : values.at(ClassID::Default);
```

Comment:
`get()` is mainly used for implementing `Layer` subclass property getters, e.g. `BackgroundLayer::getBackgroundColor()`. I intended for these to return an undefined `PropertyValue<T>` if the value wasn't set, but it looks like that's always been broken, because `values.at(ClassID::Default)` would throw in that case.

However, returning `values.at(ClassID::Fallback)` doesn't provide the desired semantics. I think the correct implementation is something like:

```
     PropertyValue<T> get() const {
         auto it = values.find(ClassID::Default);
         return it == value.end() ? PropertyValue<T>() : it->second;
     }
```